src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.exceptions.NoSuchFruitException;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.OperationType;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void doOperation(Fruit fruit) {
+        if (!FruitStorage.contains(fruit)) {
+            throw new NoSuchFruitException(OperationType.PURCHASE, fruit);
+        }
+        FruitStorage.fruits.stream()
+                .filter(x -> x.equals(fruit))
+                .findFirst().get()
+                .subtract(fruit.getQuantity());
```

Comment:
let's first check if we have enough fruit

for example, if we have in storage 49 bananas, and user wants to buy 50, we should throw exception